Rules/CTF/Scripts/CTF_GiveSpawnItems.as

```diff
@@ -222,42 +222,45 @@ void onTick(CRules@ this)
 	if ((gametime % 15) != 5)
 		return;
 
-
 	RulesCore@ core;
 	this.get("core", @core);
+	
 	if (core !is null)
 	{
-
-		CBlob@[] spots;
-		getBlobsByName(base_name(), @spots);
-		getBlobsByName("buildershop", @spots);
-		getBlobsByName("knightshop", @spots);
-		getBlobsByName("archershop", @spots);
-		for (uint step = 0; step < spots.length; ++step)
+		CBlob@ playerBlob = getLocalPlayerBlob();
+		
+		if (playerBlob !is null)
 		{
-			CBlob@ spot = spots[step];
 			CBlob@[] overlapping;
-			if (spot !is null && spot.getOverlapping(overlapping))
+			playerBlob.getOverlapping(overlapping);
+			
+			for (uint i = 0; i < overlapping.length; i++)
 			{
-				string name = spot.getName();
-				bool isShop = (name.find("shop") != -1);
-				for (uint o_step = 0; o_step < overlapping.length; ++o_step)
+				string overlappingName = overlapping[i].getName();
+				int overlappingTeam = overlapping[i].getTeamNum();
+				
+				bool isSameTeam = playerBlob.getTeamNum() == overlapping[i].getTeamNum();
+				bool isBase = false;
+				
+				for (uint b = 0; b < BASE_NAME.length; ++b)
```

Comment:
Although you will be removing this because of the previous comment, I think it's worth pointing out the following. It is a de facto standard for programmers to follow the alphabet from `i` for nested loops. In this case, you should use `j` as the index.

Reply:
Thank you.